File: Projet 1 - Qualité Air/Atelier cartographie.py
# -*- coding: utf-8 -*-
"""
Created on Fri May 24 10:47:09 2019
@author: remij

ATELIER CARTOGRAPHIE
OBJECTIF : DECOUVRIR ET SE FAMILIARISER AVEC LA VISUALISATION DE DONNEES SUR CARTES
DEROULEMENT :
- UTILISATION DE NOMINATIM, API DE GEOPY POUR CONNAITRE LES COORDONNEES D'UNE LOCALISATION CONNUE
- CHARGEMENT D'UN FOND DE CARTE OPEN STREET MAP
- AJOUT D'INFORMATIONS SUR LA CARTE

Note : cet atelier nécessite les packages 'folium' (carte OSM) et 'geopy' (geolocalisation) qui ne sont pas dans la liste de base d'Anaconda,
il faut donc les installer au préalable : Anaconda / Environnements / search 'folium' - 'geopy' puis 'Apply'
"""


# Chargement des librairies
import folium
from geopy.geocoders import Nominatim


# Requete sur PHOTON pour connaitre les coordonnees spatiales d'une adresse postale
add_MAME = "49 Boulevard Preuilly 37000 Tours France"
geocoder = Nominatim(user_agent="trainee")
geo_MAME = geocoder.geocode(add_MAME, True, 30)


# Chargement du fond de carte
carte = folium.Map(location = [geo_MAME.latitude, geo_MAME.longitude], zoom_start = 13)
print(carte)


# Ajout d'information sur la carte
folium.Marker(location = [geo_MAME.latitude, geo_MAME.longitude]).add_to(carte)
print(carte)

folium.CircleMarker(location = [geo_MAME.latitude, geo_MAME.longitude], radius = 10).add_to(carte)
print(carte)


# Personnalisation des informations restituées
msg = "DataLab 2019, c'est ici ! On s'éclate et nos formateurs sont géniaux !"
carte = folium.CircleMarker(location = [geo_MAME.latitude, geo_MAME.longitude], radius = 10, popup = msg).add_to(carte)
print(carte)


# Sauvegarde de la carte au format 'html'
import os
import webbrowser
chemin = os.getcwd() + '/carte_py.html'
carte.save(chemin)
webbrowser.open('file://' + chemin)

File: Projet 1 - Qualité Air/Atelier cartographie.R

# # ---------------------------------------------------------------------------------
# ATELIER CARTOGRAPHIE
# OBJECTIF : DECOUVRIR ET SE FAMILIARISER AVEC LA VISUALISATION DE DONNEES SUR CARTES
# DEROULEMENT :
# - UTILISATION DE PHOTON, API D'OPEN STREET MAP POUR CONNAITRE LES COORDONNEES D'UNE LOCALISATION CONNUE
# - CHARGEMENT D'UN FOND DE CARTE OPEN STREET MAP
# - AJOUT D'INFORMATIONS SUR LA CARTE
# # -----------------------------------------------------------------------------------

# Chargement des librairies
library("leaflet")
require(devtools)  
devtools::install_github(repo = 'rCarto/photon')


# Requete sur PHOTON pour connaitre les coordonnees spatiales d'une adresse postale
add_MAME <- "49 Boulevard Preuilly 37000 Tours France"
add_MAME <- gsub(" ", "+", add_MAME)
geo_MAME <- photon::geocode(add_MAME, limit = 1, key = "place")


# Chargement du fond de carte
carte <- leaflet::leaflet() %>%
  leaflet::addTiles() %>%
  leaflet::setView(lng = geo_MAME$lon, lat = geo_MAME$lat, zoom = 13)
print(carte)


# Ajout d'information sur la carte
carte <- carte %>%
  leaflet::addMarkers(geo_MAME$lon, geo_MAME$lat)
print(carte)

carte <- carte %>%
  leaflet::addCircles(lng = geo_MAME$lon, lat = geo_MAME$lat, weight = 3, radius = 100)#, color = "blue", fillColor = "blue", fillOpacity = 0.95)
print(carte)


# Personnalisation des informations restituées
msg <- "DataLab 2019, c'est ici ! On s'éclate et nos formateurs sont géniaux !"
carte <- carte %>%
  leaflet::addPopups(geo_MAME$lon, geo_MAME$lat, popup = msg, options = popupOptions(closeButton = FALSE))
print(carte)





